mne/annotations.py

```diff
@@ -331,3 +331,140 @@ def _read_annotations(fid, tree):
+        argument and return an ``int``. If string, must be 'strip-to-integer',
+        in which case it defaults to stripping event codes such as "D128" or
+        "S  1" of their non-integer parts and returns the integer.
+        If the event is not in the ``event_id`` and calling ``event_id_func``
+        on it results in a ``TypeError`` (e.g. if ``event_id_func`` is
+        ``None``) or a ``ValueError``, the event is dropped.
+    convention : str
+        Can be "eeglab" or "default". XXX
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see
+        :func:`mne.verbose` and :ref:`Logging documentation <tut_logging>`
+        for more). Defaults to self.verbose.
+
+    Returns
+    -------
+    events : ndarray, shape (n_events, 3)
+        The events.
+    event_id : dict
+        The event_id variable that can be passed to Epochs.
+    """
+    import re
+
+    assert convention in ['default', 'eeglab']
+
+    if annotations is None:
+        annotations = raw.annotations
+
+    if convention == "default":
+        inds = raw.time_as_index(
+            annotations.onset, use_rounding=True) + raw.first_samp
+
+        unique_keys = np.unique(annotations.description)
+
+        logger.info('Annotations descriptions: %s' % unique_keys)
+
+        if regexp is not None:
+            pat = re.compile(regexp)
+            unique_keys = [kk for kk in unique_keys if pat.match(kk)]
+
+        if event_id is None:
+            event_id_ = dict(
+                zip(unique_keys, np.arange(1, 1 + len(unique_keys))))
+        else:
+            event_id_ = {k: v for k, v in event_id.items()
+                         if k in unique_keys}
+
+        event_sel = [ii for ii, kk in
+                     enumerate(annotations.description)
+                     if kk in event_id_]
+        if len(event_sel) == 0:
+            raise ValueError('Could not find any of the events you specified.')
+        values = [event_id_[kk] for kk in
+                  annotations.description[event_sel]]
+        previous_value = np.zeros(len(event_sel))
+        inds = inds[event_sel]
+        events = np.c_[inds, previous_value, values].astype(int)
+        return events, event_id_
+
+    elif convention == "eeglab":
+        types = annotations.description
+        latencies = annotations.onset
+
+        if "boundary" in types and "boundary" not in event_id:
+            warn("The data contains 'boundary' events, indicating data "
+                 "discontinuities. Be cautious of filtering and epoching "
+                 "around these events.")
+
+        if len(types) < 1:  # if there are 0 events, we can exit here
+            logger.info('No events found, returning empty stim channel ...')
+            return np.zeros((0, 3)), dict()
+
+        not_in_event_id = set(x for x in types if x not in event_id)
+        not_purely_numeric = set(x for x in not_in_event_id if not x.isdigit())
```

Comment:
are these checks not needed in the default convention? Currently in default convention the event_id is created by mapping succesive unique descriptions to successive integers.